nixpkgs/nixpkgs.bzl

```diff
@@ -890,6 +890,18 @@ def nixpkgs_cc_configure_deprecated(
 
 def _nixpkgs_python_toolchain_impl(repository_ctx):
     cpu = get_cpu_value(repository_ctx)
+    os = {"darwin": "osx"}.get(cpu, "linux")
+
+    if repository_ctx.attr.target_constraints == None and repository_ctx.attr.exec_constraints == None:
+        target_constraints = ["@platforms//cpu:x86_64"]
+        target_constraints.append("@platforms//os:{}".format(os))
+        exec_constraints = target_constraints
+    else:
+        target_constraints = list(repository_ctx.attr.target_constraints)
+        exec_constraints = list(repository_ctx.attr.exec_constraints)
+
+    exec_constraints.append("@io_tweag_rules_nixpkgs//nixpkgs/constraints:support_nix")
+
```

Comment:
This is the same as in `nixpkgs_cc_configure`, correct? Could you factor this out into a function?

Reply:
with pleasure!